lib/Sema/ImportResolution.cpp

```diff
@@ -367,13 +347,19 @@ void ImportResolver::addImport(const UnboundImport &I, ModuleDecl *M) {
 // MARK: Import module loading
 //===----------------------------------------------------------------------===//
 
-ModuleDecl *
-ImportResolver::getModule(ImportPath::Module modulePath) {
+static ModuleDecl *
+getModuleImpl(ImportPath::Module modulePath, ModuleDecl *loadingModule,
+              SourceFileKind sourceFileKind = SourceFileKind::SIL) {
+  // Default is SourceFileKind::SIL to make the behavior during implicit import
+  // loading as permissive as possible.
```

Comment:
IMO this would be better as a `bool canImportBuiltinModule` parameter to avoid misusing `sourceFileKind` in the body.